tests/external_executor/executors/util.py

```diff
@@ -33,8 +53,78 @@ def terminate(self):
 
 
 @contextmanager
-def executor_thread(executor):
+def executor_thread(executor: Executor):
     et = ExecutorThread(executor)
     et.start()
     yield executor
     et.terminate()
+
+
+class ExecutorContainer:
+    def __init__(
+        self,
+        executor: str,
+        node_public_rpc_address: str,
+        network: Network,
+        supported_endpoints: Set[Tuple[str, str]],
+    ):
+        self._client = docker.DockerClient()
+
+        image_name = "ccfmsrc.azurecr.io/ccf/ci:14-02-2023-virtual"
+        LOG.info(f"Pulling image {image_name}")
+        self._client.images.pull(image_name)
+
+        # Create a container with external executor code loaded in a volume and
+        # a command to run the executor
+        command = "/workspaces/CCF/build/env/bin/python3"
```

Comment:
We should avoid hardcoding paths here as much as possible, and instead rely on the `e2e_args.workspace` variable if we can. We should also only mount what's required for the external executor to run, which is `wiki_cacher.py`, plus the generated `_pb2.files` and a `requirements.txt` file containing `loguru` and `grpc`?

Reply:
https://github.com/microsoft/CCF/pull/5018/commits/d1f01bc62346de78f95924b0af3b0c1bbe06d236

Just loads the test and infra code for now. Will do a followup PR to change the image to use something lighter which will then require the requirements file